src/o365/spo/commands/page/page-clientsidewebpart-add.ts

```diff
@@ -0,0 +1,483 @@
+			})
+			.then((res: ContextInfo): Promise<ClientSidePage> => {
+				if (this.debug) {
+					cmd.log('Response:');
+					cmd.log(res);
+					cmd.log('');
+				}
+
+				// Keep the reference of request digest for subsequent requests
+				requestDigest = res.FormDigestValue;
+
+				if (this.verbose) {
+					cmd.log(`Retrieving Client Side Page ${pageName}...`);
+				}
+				// Get Client Side Page
+				return this._getClientSidePage(pageName, cmd, args, siteAccessToken);
+			})
+			.then((csPage) => {
+				if (this.debug) {
+					cmd.log(`Retrieved Client Side Page:`);
+					cmd.log(csPage);
+					cmd.log('');
+				}
+
+				// Keep the reference of client side page for subsequent requests
+				clientSidePage = csPage;
+
+				if (this.verbose) {
+					cmd.log(
+						`Retrieving Client Side definition for WebPart ${args.options.webPartId ||
+							args.options.standardWebPart}...`
+					);
+				}
+				// Get the WebPart according to arguments
+				return this._getWebPart(cmd, args, siteAccessToken);
+			})
+			.then((webPart) => {
+				if (this.debug) {
+					cmd.log(`Retrieved WebPart definition:`);
+					cmd.log(webPart);
+					cmd.log('');
+				}
+
+				if (this.verbose) {
+					cmd.log(`Setting Client Side WebPart layout and properties...`);
+				}
+				// Set the WebPart properties and layout (section, column and order)
+				this._setWebPartPropertiesAndLayout(clientSidePage, webPart, cmd, args);
+			})
+			.then(() => {
+				if (this.verbose) {
+					cmd.log(`Saving Client Side Page...`);
+				}
+				// Save the Client Side Page with updated information
+				return this._saveClientSidePage(clientSidePage, cmd, args, siteAccessToken, requestDigest);
+			})
+			.then(() => {
+				if (this.verbose) {
+					cmd.log(
+						`Client Side WebPart ${args.options.webPartId ||
+							args.options.standardWebPart} is added to page ${pageName}`
+					);
+				}
+				cb();
+			})
+			.catch((err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+	}
+
+	private _getClientSidePage(
+		pageName: string,
+		cmd: CommandInstance,
+		args: CommandArgs,
+		accessToken: string
+	): Promise<ClientSidePage> {
+		return new Promise<ClientSidePage>((resolve, reject) => {
+			if (this.verbose) {
+				cmd.log(`Retrieving information about the page...`);
+			}
+
+			const webUrl = args.options.webUrl;
+			const requestOptions: any = {
+				url: `${webUrl}/_api/web/getfilebyserverrelativeurl('${webUrl.substr(
+					webUrl.indexOf('/', 8)
+				)}/sitepages/${encodeURIComponent(pageName)}')?$expand=ListItemAllFields/ClientSideApplicationId`,
+				headers: Utils.getRequestHeaders({
+					authorization: `Bearer ${accessToken}`,
+					'content-type': 'application/json;charset=utf-8',
+					accept: 'application/json;odata=nometadata'
+				}),
+				json: true
+			};
+
+			if (this.debug) {
+				cmd.log('Executing web request...');
+				cmd.log(requestOptions);
+				cmd.log('');
+			}
+
+			request.get(requestOptions).then((res) => {
```

Comment:
If you specify a name of the page that doesn't exist, the CLI will be stuck here, because this promise doesn't handle, rejection. As a result, the parent promise is never completed.